### PR TITLE
fix N+1

### PR DESCRIPTION
**Pull request by Kaito-Kido**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,6 +32,8 @@ gem 'pagy'
 
 gem 'sidekiq'
 
+gem 'bullet', group: 'development'
```
Comment (nathan-phan-goldenowl) [inline Gemfile:35]: put this in development group

File: app/views/header/_header.html.slim [hunk 2]
```diff
@@ -18,7 +18,7 @@
                         .ping.dot
             .div#noti-menu
                 - if current_user.notifications.first.present?
-                    = render "notifications/notifications_center", notifications: current_user.notifications.order(created_at: :desc).limit(20)
+                    = render "notifications/notifications_center", notifications: current_user.notifications.includes(sender: {avatar_attachment: :blob}).order(created_at: :desc).limit(20)
```
Comment (nathan-phan-goldenowl) [inline app/views/header/_header.html.slim:21]: put logic in controller

File: app/views/posts/show.html.slim [hunk 3]
```diff
@@ -37,7 +37,7 @@
             .submit-btn 
               =form.submit "Submit"
 
-    - @post.comments.order(created_at: :desc).each do |comment|
+    - @post.comments.includes(user: {avatar_attachment: :blob}).order(created_at: :desc).each do |comment|
```
Comment (nathan-phan-goldenowl) [inline app/views/posts/show.html.slim:40]: put logic in controller

File: app/views/posts/show.html.slim [hunk 4]
```diff
@@ -73,7 +73,7 @@
                 .dismiss-btn id="dismiss-btn-#{comment.id}" 
                   | Dismiss
       .subcomment
-        = render comment.comments
+        = render comment.comments.includes(:user, :comments)
```
Comment (nathan-phan-goldenowl) [inline app/views/posts/show.html.slim:76]: put logic in controller

File: app/jobs/create_notification_job.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateNotificationJob < ApplicationJob
+  queue_as :default
+
+  def perform(post, current_user, comment=nil)
+    if comment == nil
```
Comment (nathan-phan-goldenowl) [inline app/jobs/create_notification_job.rb:5]: Prefer using 
```
if comment
```
of 
```
if comment.present?
```

File: app/services/notification_creator_for_post_service.rb [hunk 6]
```diff
@@ -7,13 +7,15 @@ def initialize(post, current_user)
   def call
     if @post.user.admin?
       User.all.each do |user|
-        Notification.create(sender_id: @post.user.id, receiver_id: user.id, action: :post, content: "Admin has just posted an article", post_id: @post.id)
+        if user != User.admin[0]
```
Comment (nathan-phan-goldenowl) [inline app/services/notification_creator_for_post_service.rb:10]: this is bad, what if there are more than 2 admins?

Comment (Kaito-Kido) [reply app/services/notification_creator_for_post_service.rb:10]: oh no

**Review by nathan-phan-goldenowl — APPROVED** [review]

**Pull request merged**